linera-summary/src/github.rs

```diff
@@ -0,0 +1,272 @@
+            let (commit_hash, branch_name, base) = Self::get_local_git_info()?;
+            (
+                commit_hash,
+                branch_name,
+                base,
+                pr_number.ok_or_else(|| anyhow!("pr_number is None"))?,
+            )
+        } else {
+            let pr_string = env_pr_number.map_err(|_| {
+                anyhow!("GITHUB_PR_NUMBER is not set! This must be run from within CI")
+            })?;
+            (
+                env_pr_commit_hash.map_err(|_| {
+                    anyhow!("GITHUB_PR_COMMIT_HASH is not set! This must be run from within CI")
+                })?,
+                env_pr_branch.map_err(|_| {
+                    anyhow!("GITHUB_PR_BRANCH is not set! This must be run from within CI")
+                })?,
+                env_base_branch.map_err(|_| {
+                    anyhow!("GITHUB_BASE_BRANCH is not set! This must be run from within CI")
+                })?,
+                pr_string.parse().map_err(|_| {
+                    anyhow!("GITHUB_PR_NUMBER is not a valid number: {}", pr_string)
+                })?,
+            )
+        };
+
+        Ok(Self {
+            repository: GithubRepository::from_env(is_local)?,
+            pr_commit_hash,
+            pr_branch,
+            base_branch,
+            pr_number,
+        })
+    }
+}
+
+pub struct Github {
+    octocrab: Octocrab,
+    context: GithubContext,
+    is_local: bool,
+}
+
+impl Github {
+    pub fn new(is_local: bool, pr_number: Option<u64>) -> Result<Self> {
+        let octocrab_builder = Octocrab::builder();
+        let octocrab =
+            if is_local {
+                octocrab_builder
+            } else {
+                octocrab_builder.personal_token(env::var("GITHUB_TOKEN").map_err(|_| {
+                    anyhow!("GITHUB_TOKEN is not set! This must be run from within CI")
+                })?)
+            }
+            .build()
+            .map_err(|_| anyhow!("Creating Octocrab instance should not fail!"))?;
+
+        Ok(Self {
+            octocrab,
+            context: GithubContext::from_env(is_local, pr_number)?,
+            is_local,
+        })
+    }
+
+    pub fn context(&self) -> &GithubContext {
+        &self.context
+    }
+
+    pub async fn comment_on_pr(&self, body: &str) -> Result<()> {
+        if self.is_local {
+            info!("Printing summary to stdout:");
+            println!("{}", body);
+        } else {
+            info!("Commenting on PR {}", self.context.pr_number);
+            self.octocrab
+                .issues(
+                    self.context.repository.owner.clone(),
+                    self.context.repository.name.clone(),
+                )
+                .create_comment(self.context.pr_number, body)
+                .await?;
+        }
+        Ok(())
+    }
+
+    async fn latest_runs<'octo>(
+        &self,
+        branch: &str,
+        event: &str,
+        workflows_handler: &WorkflowsHandler<'octo>,
```

Comment:
Nit: not sure if this works, but if possible we could ellide the lifetime.

```suggestion
    async fn latest_runs(
        &self,
        branch: &str,
        event: &str,
        workflows_handler: &WorkflowsHandler<'_>,
```

Reply:
Nice! Yeah, that works :D